nbconvert/nbconvertapp.py

```diff
@@ -403,13 +421,17 @@ def convert_notebooks(self):
         self.exporter = exporter_map[self.export_format](config=self.config)
 
         # no notebooks to convert!
-        if len(self.notebooks) == 0:
+        if len(self.notebooks) == 0 and not self.from_stdin:
             self.print_help()
             sys.exit(-1)
 
         # convert each notebook
-        for notebook_filename in self.notebooks:
-            self.convert_single_notebook(notebook_filename)
+        if not self.from_stdin:
+            for notebook_filename in self.notebooks:
+                self.convert_single_notebook(notebook_filename)
+        else:
+            # default name when conversion from stdin
+            self.convert_single_notebook("notebook.ipynb", input_buffer=sys.stdin)
```

Comment:
Unfortunately, I think this will run into Unicode problems on Python 3 in non-UTF-8 systems, primarily Windows. Python will use the system's default encoding to try to decode stdin, and if a notebook is piped in containing a non-ascii character, and it tries to decode it using cp1252 or something, it will blow up.

Have a look at `nbconvert.utils.io.unicode_std_stream` for what we need to do here. That's for output, but basically the same steps should work for stdin.